src/Lib.P2P/PeerManager.cs

```diff
@@ -0,0 +1,211 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+
+namespace Lib.P2P
+{
+    /// <summary>
+    ///   Manages the peers.
+    /// </summary>
+    /// <remarks>
+    ///    Listens to the <see cref="SwarmService"/> events to determine the state
+    ///    of a peer.
+    /// </remarks>
+    public class PeerManager : IService
+    {
+        private static ILog _log = LogManager.GetLogger(typeof(PeerManager));
+        private CancellationTokenSource _cancel;
+
+        /// <summary>
+        ///   Initial time to wait before attempting a reconnection
+        ///   to a dead peer.
+        /// </summary>
+        /// <value>
+        ///   Defaults to 1 minute.
+        /// </value>
+        public TimeSpan InitialBackoff = TimeSpan.FromMinutes(1);
+
+        /// <summary>
+        ///   When reached, the peer is considered permanently dead.
+        /// </summary>
+        /// <value>
+        ///   Defaults to 64 minutes.
+        /// </value>
+        public TimeSpan MaxBackoff = TimeSpan.FromMinutes(64);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)